### PR TITLE
Gotchi Lending >.<

### PR DESCRIPTION
**Pull request by dudendy**

### CODE REVIEW

File: src/components/Filters/GotchiFilter.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useLocation } from 'react-router-dom';
+import { Alert, AlertTitle, Autocomplete, Avatar, Chip, Link, TextField } from '@mui/material';
+import qs from 'query-string';
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:5]: please separate, it's third party

File: src/components/Filters/GotchiFilter.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    const classes = styles();
+
+    const history = useHistory();
+    const location = useLocation();
+    const params = qs.parse(location.search);
+
+    const [guildsFilter, setGuildsFilter] = useState([]);
+
+    useEffect(() => {
+        if(params.guild) setGuildsFilter(params.guild.split(','))
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:21]: please add whitespace after `if` and wrap into `{}`

File: src/components/Filters/GotchiFilter.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:22]: no need for whitespace here

File: src/components/Filters/GotchiFilter.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        if(!dataLoading) filterGotchis(guildsFilter);
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:27]: please add whitespace after `if` and wrap into `{}`

File: src/components/Filters/GotchiFilter.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:28]: no need for whitespace here

File: src/components/Filters/GotchiFilter.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [dataLoading]);
+
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:31]: extra whitespace

File: src/components/Filters/GotchiFilter.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    const handleChange = (event, newValue) => {
+        filterGotchis(newValue);
+        handleQueryParams(newValue)
+        setGuildsFilter(newValue);
+    };
+
+    const filterGotchis = (filters) => {
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:39]: please pass `gotchis` as param here

File: src/components/Filters/GotchiFilter.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        const filtered = gotchis.filter((gotchi) => {
+            return filters.some((filter) => filter === gotchi.guild);
+        });
+
+        setGotchis(filters.length > 0 ? filtered : gotchis);
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:44]: a little bit bad approach here. You're doing some filtering and after that checking some condition. Will be better to check at first and after that filter if needed

File: src/components/Filters/GotchiFilter.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+    const handleChange = (event, newValue) => {
+        filterGotchis(newValue);
+        handleQueryParams(newValue)
+        setGuildsFilter(newValue);
+    };
+
+    const filterGotchis = (filters) => {
+        const filtered = gotchis.filter((gotchi) => {
+            return filters.some((filter) => filter === gotchi.guild);
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:41]: no need for `return` here, just do not wrap into `{}`

File: src/components/Filters/GotchiFilter.js [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+                    renderInput={(params) => (
+                        <TextField
+                            {...params}
+                            size='small'
+                            label='Guilds'
+                        />
+                    )}
+                    renderTags={(tagValue, getTagProps) =>
+                        tagValue.map((option, index) => (
+                            <Chip
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:76]: not really sure, but maybe here would be good to add `key` prop

File: src/components/Filters/GotchiSorting.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    }
+];
+
+export default function GotchiSorting({ gotchis, setGotchis, sortings, defaultSorting }) {
+    const classes = GotchiSortingStyles();
+    const [sorting, setSorting] = useState(defaultSorting);
+    const [direction, setDirection] = useState('desc');
+
+    const onSortChange = (event, value) => {
+        if (!value) return;
```
Comment (dmutrokot) [inline src/components/Filters/GotchiSorting.js:36]: please wrap into `{}`

File: src/components/Filters/GotchiSorting.js [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+        setSorting(value);
+        sortGotchis(value, direction);
+    };
+
+    const onDirectionChange = (event, value) => {
+        if (!value) return;
```
Comment (dmutrokot) [inline src/components/Filters/GotchiSorting.js:43]: please wrap into `{}`

File: src/components/Gotchi/GotchiLending/GotchiLending.js [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+    const classes = styles();
+
+    const periodToMillis = gotchi.period * 1000;
+    const periodObject = Duration.fromMillis(periodToMillis).shiftTo('years', 'months', 'days', 'hours').normalize().toObject();
+
+    const renderPeriod = () => {
+        return (
+            <>
+                {Object.entries(periodObject).map((period, index) => {
+                    let value = period[1];
```
Comment (dmutrokot) [inline src/components/Gotchi/GotchiLending/GotchiLending.js:25]: please use `const`

File: src/components/Gotchi/GotchiLending/GotchiLending.js [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+                    let key = value > 1 ? period[0] : period[0].slice(0, -1);
```
Comment (dmutrokot) [inline src/components/Gotchi/GotchiLending/GotchiLending.js:26]: please use `const`

File: src/components/Gotchi/GotchiLending/GotchiLending.js [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+
+                    if(value === 0) return null;
```
Comment (dmutrokot) [inline src/components/Gotchi/GotchiLending/GotchiLending.js:28]: please add space after `if` and wrap into `{}`

File: src/components/Gotchi/GotchiLending/GotchiLending.js [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+                        <span className={gotchi.splitOther > 0 ? 'highlight' : undefined}>
+                            {gotchi.splitOther}
+                        </span>%
+                    </div>
+                </div>
+            </div>
+
+            <div className={classNames(classes.section, classes.tokens)}>
+                {gotchi.tokensToShare.map((token, index) => {
+                    let tokenName = graphUtils.getTokenName(token);
```
Comment (dmutrokot) [inline src/components/Gotchi/GotchiLending/GotchiLending.js:80]: please use `const`

File: src/components/Lazy/GotchisLazy.js [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+    display: grid;
+    grid-template-columns: repeat(auto-fill,minmax(192px,1fr));
+    grid-gap: 12px;
+`;
+
+export default function GotchisLazy({ items, render }) {
+    const gridRef = useRef(null);;
+
+    useEffect(() => {
+        if(items.length) scrollToTop();
```
Comment (dmutrokot) [inline src/components/Lazy/GotchisLazy.js:18]: please add space after `if` and wrap into `{}`

File: src/components/Lazy/GotchisLazy.js [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+    const scrollToTop = () => {
+        gridRef.current.scrollToIndex({
+            index: 0,
+            align: 'start',
+            behavior: 'auto'
+        });
+    };
+
+
+    if (!items) return;
```
Comment (dmutrokot) [inline src/components/Lazy/GotchisLazy.js:30]: please wrap into `{}`

File: src/components/Lazy/GotchisLazy.js [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+
+    if (items.length === 0) return (
```
Comment (dmutrokot) [inline src/components/Lazy/GotchisLazy.js:32]: please wrap into `{}`

File: src/pages/Lend/Lend.js [hunk 20]
```diff
@@ -0,0 +1,143 @@
...
+];
+
+export default function Lend() {
+    const [lendings, setLendings] = useState([]);
+    const [lendingsCache, setLendingsCache] = useState([]);
+    const [dataLoading, setDataLoading] = useState(true);
+
+    const defaultSorting = 'timeCreated';
+
+    const availableGuilds = guilds.filter((guild) => {
```
Comment (dmutrokot) [inline src/pages/Lend/Lend.js:79]: `(guild)` no need for brackets and `return` statement

File: src/pages/Lend/Lend.js [hunk 21]
```diff
@@ -0,0 +1,143 @@
...
+        return guild.members.length > 0;
+    });
+    const guildsKeys = availableGuilds.map((guild)=> {
```
Comment (dmutrokot) [inline src/pages/Lend/Lend.js:82]: `(guild)` no need for brackets and `return` statement

File: src/utils/gotchiverseUtils.js [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+        try {
+            return require(`../assets/images/guilds/${name}.png`).default;
+        } catch (error) {
+            return require(`../assets/images/image-placeholder.svg`).default;
+        }
+    },
+
+    gedAddressGuild(address) {
+        const guild = guilds.filter((guild) => {
+            return guild.members.some((member) => member.toLowerCase() === address.toLowerCase())
```
Comment (dmutrokot) [inline src/utils/gotchiverseUtils.js:17]: no need for `return`

**Review by dmutrokot — CHANGES_REQUESTED** [review]

File: src/components/Lazy/GotchisLazy.js (hunk 18 above)
Comment (dudendy) [reply src/components/Lazy/GotchisLazy.js:30]: please let me do this magic ^^

File: src/components/Lazy/GotchisLazy.js (hunk 19 above)
Comment (dudendy) [reply src/components/Lazy/GotchisLazy.js:32]: please let me do this magic ^^

File: src/components/Filters/GotchiFilter.js (hunk 7 above)
Comment (dudendy) [reply src/components/Filters/GotchiFilter.js:40]: that's make no sense

File: src/components/Filters/GotchiFilter.js (hunk 8 above)
Comment (dudendy) [reply src/components/Filters/GotchiFilter.js:45]: solved

File: src/components/Filters/GotchiFilter.js (hunk 10 above)
Comment (dudendy) [reply src/components/Filters/GotchiFilter.js:77]: not needed here

File: src/components/Filters/GotchiFilter.js [hunk 23]
```diff
@@ -0,0 +1,114 @@
...
+
+    useEffect(() => {
+        if(params.guild) {
+            setGuildsFilter(params.guild.split(','));
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+        if(!dataLoading) {
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:29]: space missed

File: src/components/Filters/GotchiFilter.js [hunk 24]
```diff
@@ -0,0 +1,114 @@
...
+    const classes = styles();
+
+    const history = useHistory();
+    const location = useLocation();
+    const params = qs.parse(location.search);
+
+    const [guildsFilter, setGuildsFilter] = useState([]);
+
+    useEffect(() => {
+        if(params.guild) {
```
Comment (dmutrokot) [inline src/components/Filters/GotchiFilter.js:22]: space missed

File: src/components/Gotchi/GotchiLending/GotchiLending.js [hunk 25]
```diff
@@ -0,0 +1,99 @@
...
+    const periodObject = Duration.fromMillis(periodToMillis).shiftTo('years', 'months', 'days', 'hours').normalize().toObject();
+
+    const renderPeriod = () => {
+        return (
+            <>
+                {Object.entries(periodObject).map((period, index) => {
+                    const value = period[1];
+                    const key = value > 1 ? period[0] : period[0].slice(0, -1);
+
+                    if(value === 0) {
```
Comment (dmutrokot) [inline src/components/Gotchi/GotchiLending/GotchiLending.js:28]: space missed

File: src/components/Lazy/GotchisLazy.js [hunk 26]
```diff
@@ -0,0 +1,62 @@
...
+const NoContent = styled.div`
+    text-align: center;
+    padding: 24px;
+`;
+
+export default function GotchisLazy({ items, render }) {
+    const gridRef = useRef(null);;
+
+    useEffect(() => {
+        if(items.length) {
```
Comment (dmutrokot) [inline src/components/Lazy/GotchisLazy.js:23]: space missed

File: src/pages/Lend/Lend.js [hunk 27]
```diff
@@ -0,0 +1,139 @@
...
+export default function Lend() {
+    const [lendings, setLendings] = useState([]);
+    const [lendingsCache, setLendingsCache] = useState([]);
+    const [dataLoading, setDataLoading] = useState(true);
+
+    const defaultSorting = 'timeCreated';
+
+    const availableGuilds = guilds.filter((guild) => guild.members.length > 0);
+    const guildsKeys = availableGuilds.map((guild) => commonUtils.stringToKey(guild.name));
+
```
Comment (dmutrokot) [inline src/pages/Lend/Lend.js:81]: extra

**Review by dmutrokot — APPROVED** [review]

**Pull request merged**